### DISCUSSION TITLE
Configuring SL4J Logging

### DISCUSSION THREAD

**Discussion by andreiyusupau** [quickfix-j/quickfixj#866] [Q&A] [answered]
I have configured my quickfix j initiator to use SL4J for logging with following settings:
```
SLF4JLogEventCategory=DEBUG
SLF4JLogErrorEventCategory=ERROR
SLF4JLogIncomingMessageCategory=DEBUG
SLF4JLogOutgoingMessageCategory=DEBUG
SLF4JLogPrependSessionID=Y
SLF4JLogHeartbeats=N
```
So I've expected that only errors would be logged if my root log level is `INFO`.
But for some reasons messages like this are still logged:
`13:47:54.286 INFO  [main] DEBUG - FIX.4.4:xxx->yyy: Created session: FIX.4.4:xxx->yyy`
Could it be some misconfiguration?

**Comment by chrjohn** [comment]
Hi @andreiyusupau ,

a category is not the same as the level. You could categorize log messages into categories, see https://github.com/quickfix-j/quickfixj/blob/master/quickfixj-core/src/main/java/quickfix/SLF4JLog.java

> only errors would be logged if my root log level is `INFO`

Actually, this will also log `INFO` and events are logged at INFO level, see https://github.com/quickfix-j/quickfixj/blob/36a07fea4bf104a251172c7dd835241b22604508/quickfixj-core/src/main/java/quickfix/SLF4JLog.java#L114 and https://github.com/quickfix-j/quickfixj/blob/36a07fea4bf104a251172c7dd835241b22604508/quickfixj-core/src/main/java/quickfix/SLF4JLog.java#L135

**Reply by andreiyusupau** [reply]
Thanks for fast reply.
So if I understand correctly, I can't affect log levels of messages using settings (`Created session: ...` for example would be always `INFO`). And the only way to hide message logs for production environment would be setting higher level for `quickfix` package (for example `ERROR`)?

**Reply by chrjohn** [reply, marked as answer]
Correct, you cannot influence it using settings. You could however implement/override your own `Log` to implement a custom log method which could log to another level.
